packagehandlers/gradlepackagehandler.go

```diff
@@ -0,0 +1,185 @@
+)
+
+var regexpPatterns []string
+
+func init() {
+	// Example: <groupName>:<packageName>:<version>
+	regexpPatterns = append(regexpPatterns, directStringWithVersionRegexp)
+
+	groupEntry := getMapRegexpEntry("group")
+	nameEntry := getMapRegexpEntry("name")
+	versionEntry := getMapRegexpEntry("version")
+
+	// Example: group: "<groupName>", name: "<packageName>", version: "<version>" | group = "<groupName>", name = "<packageName>", version = "<version>"
+	directMapRegexpPattern := groupEntry + regexpMapValueWithApostrophes + "," + nameEntry + regexpMapValueWithApostrophes + "," + versionEntry + regexpMapValueWithApostrophes
+	regexpPatterns = append(regexpPatterns, directMapRegexpPattern)
+}
+
+type GradlePackageHandler struct {
+	CommonPackageHandler
+}
+
+func (gph *GradlePackageHandler) UpdateDependency(vulnDetails *utils.VulnerabilityDetails) error {
+	if vulnDetails.IsDirectDependency {
+		return gph.updateDirectDependency(vulnDetails)
+	}
+
+	return &utils.ErrUnsupportedFix{
+		PackageName:  vulnDetails.ImpactedDependencyName,
+		FixedVersion: vulnDetails.SuggestedFixedVersion,
+		ErrorType:    utils.IndirectDependencyFixNotSupported,
+	}
+}
+
+func (gph *GradlePackageHandler) updateDirectDependency(vulnDetails *utils.VulnerabilityDetails) (err error) {
+	if !isVersionSupportedForFix(vulnDetails.ImpactedDependencyVersion) {
+		return &utils.ErrUnsupportedFix{
+			PackageName:  vulnDetails.ImpactedDependencyName,
+			FixedVersion: vulnDetails.SuggestedFixedVersion,
+			ErrorType:    utils.UnsupportedForFixVulnerableVersion,
+		}
+	}
+
+	// A gradle project may contain several descriptor files in several sub-modules. Each vulnerability may be found in each of the descriptor files.
+	// Therefore we iterate over every descriptor file for each vulnerability and try to find and fix it.
+	descriptorFilesPaths, err := getDescriptorFilesPaths()
+	if err != nil {
+		return
+	}
+
+	for _, descriptorFilePath := range descriptorFilesPaths {
+		err = fixVulnerabilityIfExists(descriptorFilePath, vulnDetails)
+		if err != nil {
+			return
+		}
+	}
+
+	return
+}
+
+// isVersionSupportedForFix checks if the impacted version is currently supported for fix
+func isVersionSupportedForFix(impactedVersion string) bool {
+	if strings.Contains(impactedVersion, "+") ||
+		(strings.Contains(impactedVersion, "[") || strings.Contains(impactedVersion, "(")) ||
+		strings.Contains(impactedVersion, "latest.release") {
+		return false
+	}
+	return true
+}
+
+// getDescriptorFilesPaths collects all descriptor files absolute paths
+func getDescriptorFilesPaths() (descriptorFilesPaths []string, err error) {
+	err = filepath.WalkDir(".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return fmt.Errorf("error has occured when trying to access or traverse the files system: %s", err.Error())
+		}
+
+		if d.Type().IsRegular() && (strings.HasSuffix(path, groovyBuildFileSuffix) || strings.HasSuffix(path, kotlinBuildFileSuffix)) {
```

Comment:
![Screenshot 2023-09-27 at 9 44 36](https://github.com/jfrog/frogbot/assets/81029514/f13980d8-4788-406f-89be-f56513efe583)

for any build file that is not in the main directory we will get the relative path to it and it will not be equal to build.gradle or build.gradle.kts exactly